citesphere-importer/src/main/java/edu/asu/diging/citesphere/importer/core/service/parse/impl/WoSHandler.java

```diff
@@ -27,6 +33,19 @@
     public boolean canHandle(String path) throws HandlerTestException {
         File file = new File(path);
         
+        try {
+            BOMInputStream inputStream;
+            inputStream = new BOMInputStream(FileUtils.openInputStream(file), false, ByteOrderMark.UTF_8, 
+                    ByteOrderMark.UTF_16BE, ByteOrderMark.UTF_16LE, ByteOrderMark.UTF_32BE, ByteOrderMark.UTF_32LE);
+            String content;
+            if (inputStream.hasBOM()) {
+                content = IOUtils.toString(inputStream, inputStream.getBOMCharsetName());
+                FileUtils.write(file, content, "UTF8");
```

Comment:
hm, this load the whole file into memory and writes it out all at once, right? this could lead to issues if someone uploads a very big file. you might have to write it into another file line by line and then replace the old file.